ludwig/models/gbm.py

```diff
@@ -94,75 +94,26 @@ def forward(
         ],
         mask=None,
     ) -> Dict[str, torch.Tensor]:
-        if self.compiled_model is None:
-            raise ValueError("Model has not been trained yet.")
-
-        if isinstance(inputs, tuple):
-            inputs, targets = inputs
-            # Convert targets to tensors.
-            for target_feature_name, target_value in targets.items():
-                if not isinstance(target_value, torch.Tensor):
-                    targets[target_feature_name] = torch.from_numpy(target_value)
-                else:
-                    targets[target_feature_name] = target_value
-        else:
-            targets = None
-
-        assert list(inputs.keys()) == self.input_features.keys()
-
-        # Convert inputs to tensors of type float as expected by hummingbird GEMMTreeImpl.
-        for input_feature_name, input_values in inputs.items():
-            if not isinstance(input_values, torch.Tensor):
-                inputs[input_feature_name] = torch.from_numpy(input_values).float()
-            else:
-                inputs[input_feature_name] = input_values.view(-1, 1).float()
-
-        # TODO(travis): include encoder and decoder steps during inference
-        # encoder_outputs = {}
-        # for input_feature_name, input_values in inputs.items():
-        #     encoder = self.input_features[input_feature_name]
-        #     encoder_output = encoder(input_values)
-        #     encoder_outputs[input_feature_name] = encoder_output
-
-        # concatenate inputs
-        inputs = torch.cat(list(inputs.values()), dim=1)
-
         # Invoke output features.
         output_logits = {}
         output_feature_name = self.output_features.keys()[0]
         output_feature = self.output_features[output_feature_name]
 
-        assert (
-            type(inputs) is torch.Tensor
-            and inputs.dtype == torch.float32
-            and inputs.ndim == 2
-            and inputs.shape[1] == len(self.input_features)
-        ), (
-            f"Expected inputs to be a 2D tensor of shape (batch_size, {len(self.input_features)}) of type float32, "
-            f"but got {inputs.shape} of type {inputs.dtype}"
-        )
-        # Predict using PyTorch module, so it is included when converting to TorchScript.
-        preds = self.compiled_model.model(inputs)
+        if isinstance(inputs, tuple):
+            inputs, _ = inputs
+
+        assert list(inputs.keys()) == self.input_features.keys()
+
+        in_array = np.stack(list(inputs.values()), axis=0).T
 
         if output_feature.type() == NUMBER:
-            # regression
+            preds = torch.from_numpy(self.lgbm_model.predict(in_array))
```

Comment:
Quick thing I've seen in the past: Torch sometimes creates warnings when we create a Tensor from the array directly rather than a copy of the array, so you may want to change these to:

``` 
preds = torch.from_numpy(np.array(self.lgbm_model.predict(in_array)), copy=True)
```

Reply:
Ah, good catch! I don't think I encountered that in the benchmark, but will run a quick test to see if copying has any significant overhead.